nixos/modules/services/misc/nix-optimise.nix

```diff
@@ -43,14 +33,19 @@ in
       }
     ];
 
-    systemd.services.nix-optimise = lib.mkIf config.nix.enable
-      { description = "Nix Store Optimiser";
+    systemd = lib.mkIf config.nix.enable {
+      services.nix-optimise = {
+        description = "Nix Store Optimiser";
         # No point this if the nix daemon (and thus the nix store) is outside
         unitConfig.ConditionPathIsReadWrite = "/nix/var/nix/daemon-socket";
         serviceConfig.ExecStart = "${config.nix.package}/bin/nix-store --optimise";
-        startAt = optionals cfg.automatic cfg.dates;
+        startAt = lib.optionals cfg.automatic cfg.dates;
       };
 
+      timers.nix-optimise.timerConfig = {
+        Persistent = true;
+        RandomizedDelaySec = 1800;
```

Comment:
Why do you hardcode `RandomizedDelaySec`?

It is also not mentioned in the commit message.

Reply:
The default timer setting for this module runs daily and randomizing the exact runtime by 30 minutes prevents trickle down effects. Probably every timer wants this setting in some form otherwise it could create load spikes whenever multiples are run at the same time.
Maybe we can set this to 10 min instead, I don't think that matters to much. Not sure if we want a setting for this, feels more like ~~feature~~ settings creep.